src/lib.rs

```diff
@@ -429,11 +434,19 @@ pub fn publish_package_response(response: http::Response<Vec<u8>>) -> Result<(),
     // TODO: return data from body
     let (parts, body) = response.into_parts();
     match parts.status {
+        StatusCode::OK => Ok(()),
         StatusCode::CREATED => Ok(()),
         StatusCode::NOT_FOUND => Err(ApiError::NotFound),
         StatusCode::TOO_MANY_REQUESTS => Err(ApiError::RateLimited),
         StatusCode::UNAUTHORIZED => Err(ApiError::InvalidApiKey),
         StatusCode::FORBIDDEN => Err(ApiError::Forbidden),
+        StatusCode::UNPROCESSABLE_ENTITY => {
+            let error: serde_json::Value =
+                serde_json::from_str(&String::from_utf8_lossy(&body).to_string())?;
+            Err(ApiError::InvalidModification(
+                error["errors"]["inserted_at"].to_string(),
```

Comment:
This can panic if the format isn't what we expect. Let's use a safe way of introspecting this such as parsing to a struct or just looking for a substring.